.github/workflows/workflow-build.yml

```diff
@@ -122,12 +123,31 @@ jobs:
         run: make otelcol-sumo-${{inputs.arch_os}}
         working-directory: ./otelcolbuilder
 
-      - name: Build (FIPS)
+      - uses: actions/cache/restore@v3
+        id: restore-toolchain-cache
         if: ${{ inputs.fips && contains(inputs.arch_os, 'linux') && (! inputs.only-if-changed || steps.changed-files.outputs.any_changed == 'true') }}
-        id: containerized-build
-        uses: ./ci/build-fips-action
         with:
-          go-version: ${{ env.GO_VERSION }}
+          path: |
+            /opt/toolchain
+          key: ${{ steps.get-cache-key.outputs.toolchain-cache-key }}
+
+      - name: Rebuild Toolchains
+        id: rebuild-toolchain
+        if: ${{ steps.restore-toolchain-cache.outcome == 'success' && steps.restore-toolchain-cache.outputs.cache-hit != 'true' }}
+        run: make -C ./ci/build-fips-action toolchain-${{ inputs.arch_os }} OUTPUT=/opt/toolchain -j3
+
+      - name: Build (FIPS)
+        if: ${{ inputs.fips && contains(inputs.arch_os, 'linux') && (! inputs.only-if-changed || steps.changed-files.outputs.any_changed == 'true') }}
+        run: |
+          CC=$(find /opt/toolchain/bin -type f -name "*-linux-musl-gcc")
+          test "$CC"
+          echo "Using toolchain: $CC"
+          make otelcol-sumo-${{inputs.arch_os}} \
+            FIPS_SUFFIX="-fips" \
+            CGO_ENABLED="1" \
+            CC="$CC" \
+            EXTRA_LDFLAGS="-linkmode external -extldflags '-static'"
```

Comment:
Could we add this as a target to the `otelcolbuilder` Makefile? I'd like this build to be relatively easy to do locally, and currently we're a bit too coupled to Github Actions.